frontend/src/employee-frontend/components/holiday-term-periods/TermsSection.tsx

```diff
@@ -44,53 +103,104 @@ export default React.memo(function TermsSection() {
-                <Tr key={i} data-qa="preschool-term-row">
-                  <Td data-qa="finnish-preschool">
-                    {row.finnishPreschool.format('dd.MM.yyyy')}
-                  </Td>
-                  {featureFlags.extendedPreschoolTerm && (
-                    <Td data-qa="extended-term-start">
-                      {row.extendedTerm.start.format('dd.MM.yyyy')}
+        <>
+          <Table>
+            <Thead>
+              <Tr>
+                <Th>{i18n.terms.finnishPreschool}</Th>
+                {featureFlags.extendedPreschoolTerm && (
+                  <Th>{i18n.terms.extendedTermStart}</Th>
+                )}
+                <Th>{i18n.terms.applicationPeriodStart}</Th>
+                <Th>{i18n.terms.termBreaks}</Th>
+                <Th />
+              </Tr>
+            </Thead>
+            <Tbody>
+              {preschoolTerms
+                .sort((a, b) =>
+                  b.finnishPreschool.start.compareTo(a.finnishPreschool.start)
+                )
+                .map((row, i) => (
+                  <Tr key={i} data-qa="preschool-term-row">
+                    <Td data-qa="finnish-preschool">
+                      {row.finnishPreschool.format('dd.MM.yyyy')}
+                    </Td>
+                    {featureFlags.extendedPreschoolTerm && (
+                      <Td data-qa="extended-term-start">
+                        {row.extendedTerm.start.format('dd.MM.yyyy')}
+                      </Td>
+                    )}
+                    <Td data-qa="application-period-start">
+                      {row.applicationPeriod.start.format('dd.MM.yyyy')}
+                    </Td>
+                    <Td data-qa="term-breaks">
+                      <Ul>
+                        {row.termBreaks.map((termBreak, i) => (
+                          <li
+                            data-qa={`term-break-${termBreak.start.formatIso()}`}
+                            key={`term-break-${i}`}
+                          >
+                            {termBreak.formatCompact()}
+                          </li>
+                        ))}
+                      </Ul>
+                    </Td>
+                    <Td>
+                      <FixedSpaceRow spacing="s">
+                        <IconButton
+                          icon={faPen}
+                          data-qa="btn-edit"
+                          onClick={() => onEditTermHandle(row)}
+                          aria-label={i18n.common.edit}
+                        />
+                        {row.finnishPreschool.start.isAfter(
+                          LocalDate.todayInSystemTz()
+                        ) && (
+                          <IconButton
+                            icon={faTrash}
+                            data-qa="btn-delete"
+                            onClick={() => onDeletePreschoolTerm(row.id)}
+                            aria-label={i18n.common.edit}
+                          />
```

Comment:
FWW, this thing could be done with much less code using a semi-recent component called ConfirmedMutation. You just pass in the texts and the mutation to be run if confirmed. You don't need to handle any state and modal rendering etc yourself.

Reply:
I did not realize that this ConfirmedMutation was a thing - great point and it made the implementation much cleaner!